### ISSUE TITLE
Generic MicrosoftAzure error: At least one authorization option must be specified

### ISSUE THREAD

**Issue by mark25913** [delta-io/delta-rs#838]
# Environment

**Delta-rs version**: 0.6.0

**Binding**: python version 3.9.8

**Environment**:
- **Cloud provider**: Azure storage
- **OS**: MacOS Monterey, version 12.6
- **Other**: 0.6.1 is not supported for MacOS Monterey

***
# Bug

**What happened**:
It worked within followed code:
```
from deltalake import DeltaTable

storage_options = {
    "AZURE_STORAGE_ACCOUNT_KEY"= "xxxxxxxxxxxxxxxx",
    "AZURE_STORAGE_ACCOUNT_NAME"= "xxxxxxxxxxxxxxxx"
}
table_uri = abfs[s]://<file_system>@<account_name>.dfs.core.windows.net/<path>/<file_name>
dt = DeltaTable(table_uri, storage_options=storage_options)

schema = dt.schema().to_pyarrow() # It worked successfully.
files = dt.files() # It worked successfully.
```
But it doesn't work within followed code:
```
df = dt.to_pandas() # It doesn't work.
dataset = dt.to_pyarrow_dataset() # It doesn't work.
```
The error log:
```
Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/pydevd.py", line 1496, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/my_folder_path/delta_test.py", line 20, in <module>
    data = dt.to_pandas()
  File "/my_folder_path/.pyenv/versions/delta.rs-venv/lib/python3.9/site-packages/deltalake/table.py", line 347, in to_pandas
    return self.to_pyarrow_table(
  File "/my_folder_path/.pyenv/versions/delta.rs-venv/lib/python3.9/site-packages/deltalake/table.py", line 329, in to_pyarrow_table
    return self.to_pyarrow_dataset(
  File "/my_folder_path/.pyenv/versions/delta.rs-venv/lib/python3.9/site-packages/deltalake/table.py", line 295, in to_pyarrow_dataset
    DeltaStorageHandler(self._table.table_uri())
  File "/my_folder_path/.pyenv/versions/delta.rs-venv/lib/python3.9/site-packages/deltalake/fs.py", line 15, in __init__
    self._storage = DeltaStorageFsBackend(table_uri)
deltalake.PyDeltaTableError: Failed to read delta log object: Generic MicrosoftAzure error: At least one authorization option must be specified
```

**What you expected to happen**:
I expected that the second code can work through the same DeltaTable instance.

**How to reproduce it**:
Use the storage_options parameter to connect azure backends.

**More details**:

**Comment by roeap** [comment]
Could you try setting the options in the environment as well? Right now the storage configuration does not get properly passed trough on all code paths. The fix is already on master and will be available in the next release.

**Comment by mark25913** [comment]
Thanks for your reply, it seem to work. However I got another error message:
```
deltalake.PyDeltaTableError: Object at location abfs[s]://<file_system>@<account_name>.dfs.core.windows.net/<path>/<file_name> not found: response error "﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>BlobNotFound</Code><Message>The specified blob does not exist.
RequestId:xxxxxxxxxxxxxxxxx
Time:2022-09-25T06:28:44.6321252Z</Message></Error>", after 0 retries: HTTP status client error (404 Not Found) for url (https://my_blob_file_path.parquet)
```

I expected the result of function "DeltaTable.to_pandas()" will be the same as function "spark.read.format('delta').load(my_data_path).show()" which is the python package delta-spark. I got the data using the function of delta-spark when I used the same data path of azure storage blob for testing the difference result of delta.rs V.S. delta-spark.

Did you have a recommended solution for this error message? I hope that the solution will be not to modify the original configuration of azure storage (like public access level etc).

I am looking forward to your reply.

**Comment by roeap** [comment]
The urls in the error message are quite confusing, I guess they were edited. The message is a 404 - not found. So either the url is wrong, or no object exists at that location. We are getting a response from the server though.

In any case, we recommend using much simpler uri notation. e.g. `az://<container>/<path>`, where <container> is the container in your account, and <path> the oath to your delta table root.

**Comment by mark25913** [comment]
Thanks for your reply and recommendation. You are right, I edited the urls that I would not like to show the real storage blob url path. However, I would like to confirm why I got the 404 message?

My test situation:
I can get storage blob and data with my storage blob url path when I use delta-spark function to query data, but I cannot get  blob and data with the same storage blob url path when I use delta.rs function.
